### ISSUE TITLE
✨ Add PM-01, 02, 03 & 04 as classification options on the Search page

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#9223]
## ✨ Feature

Add PM-01 through -04 as classification filter options on the Search page, along with IT-01 through -04.

**EN**
1. PM-01: Junior Officer
2. PM-02: Junior Analyst
3. PM-03: Officer
4. PM-04: Analyst

**FR**
1. PM-01: Agent(e) subalterne
2. PM-02: Analyste subalterne
3. PM-03: Agent(e)
4. PM-04: Analyste

## 📸 Screenshot

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4313717/09385c57-cfb3-4434-bdb1-8571fddef2aa)

## ✅ Acceptance Criteria

- [ ] Users can filter by PM classifications (01-04) on the Find Talent page
- [ ] While we're at it, remove the salary range values on the IT classification options

**Comment by tristan-orourke** [comment]
@esizer: it should be easy to autoselect [on s if that is what we want to do. Should just be a setValue after setting classification

**Comment by mnigh** [comment]
@tristan-orourke some questions:

1. By working on this issue, does this mean that https://github.com/GCTC-NTGC/gc-digital-talent/issues/7601 can be closed as well?
2. Should this issue be blocked by https://github.com/GCTC-NTGC/gc-digital-talent/issues/9160 since that issue deals with changing the scope used in this issue?
3. The `inItPublishingGroup` scope restricts "a query by pool candidates that are for pools containing IT specific publishing groups" (`IT_JOBS_ONGOING`, `IT_JOBS`): are there other publishing groups that need to added that processes (formerly called pools) using PM classifications require?
4. What is the need for conditional options in streams based on the classification selection since when creating a process, any stream can be selected? Additionally, the search page allows for selection of a classification of **IT-01** with a stream of **Executive Group** that would technically yield candidates but practically, probably should not but is left to the user to make searches on their own. Also, hardcoding these kind of conditions seems like it will be increasingly more difficult to understand as the conditionals become more complex.

**Comment by tristan-orourke** [comment]
1. Yes, for sure
2. No, don't wait
3. I misunderstood this scope! Scoping by _publishing group_ and not classification is actually fine.
4. Ughhh... You're right. Let's move those criteria to a new ticket.

**Comment by tristan-orourke** [comment]
@gobyrne we have some old logic on the search page which only shows Classifications which have at least one published pool. (This is in addition to restricting it to IT 1-4). Do you want us to keep that logic for the PM classification (not sure how many PM pools have been published yet), or strip out that logic entirely and just show PM 1-4 right away.

**Comment by gobyrne** [comment]
Doesn't matter to me. There is exactly one published pool in each of PM-01, 2, 3, 4
